### PR TITLE
Add mapping codes for the wiimote

### PR DESCRIPTION
**Pull request by tirithen**
## Description
Add mappings for wiimote keys.

## Motivation and Context
In multiple forums people are having problems with the wiimote in kodi, namely that only the d-poad works but it's not possible to use A/B/-/Home/+/1/2 buttons. The logs when pushing these buttons gives the following:

```
22:03:45.696 T:1958289408   DEBUG: Previous line repeats 1 times.
22:03:45.696 T:1958289408   DEBUG: CLinuxInputDevice::KeyEvent: TranslateKey returned XBMCK_UNKNOWN from code(304)
22:03:47.130 T:1958289408   DEBUG: Previous line repeats 1 times.
22:03:47.130 T:1958289408   DEBUG: CLinuxInputDevice::KeyEvent: TranslateKey returned XBMCK_UNKNOWN from code(305)
22:03:49.130 T:1958289408   DEBUG: Previous line repeats 1 times.
22:03:49.130 T:1958289408   DEBUG: CLinuxInputDevice::KeyEvent: TranslateKey returned XBMCK_UNKNOWN from code(412)
22:03:50.230 T:1958289408   DEBUG: Previous line repeats 1 times.
22:03:50.230 T:1958289408   DEBUG: CLinuxInputDevice::KeyEvent: TranslateKey returned XBMCK_UNKNOWN from code(316)
22:03:51.080 T:1958289408   DEBUG: Previous line repeats 1 times.
22:03:51.080 T:1958289408   DEBUG: CLinuxInputDevice::KeyEvent: TranslateKey returned XBMCK_UNKNOWN from code(407)
22:03:52.513 T:1958289408   DEBUG: Previous line repeats 1 times.
22:03:52.513 T:1958289408   DEBUG: CLinuxInputDevice::KeyEvent: TranslateKey returned XBMCK_UNKNOWN from code(257)
22:03:53.563 T:1958289408   DEBUG: Previous line repeats 1 times.
22:03:53.564 T:1958289408   DEBUG: CLinuxInputDevice::KeyEvent: TranslateKey returned XBMCK_UNKNOWN from code(258)

``` 
The conclusion from several of the forum threads is that those codes was not mapped in LinuxInputDevices.cpp, hence this change.

First time contributing here, so tell me if something more is needed.

## Types of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] Improvement (non-breaking change which improves existing functionality)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### CODE REVIEW

File: xbmc/input/linux/LinuxInputDevices.cpp [hunk 1]
```diff
@@ -280,6 +280,14 @@ KeyMap keyMap[] = {
   { 378               , XBMCK_RIGHT       }, // Green
   { 381               , XBMCK_UP          }, // Yellow
   { 366               , XBMCK_DOWN        }, // Blue
+  // Wiimote controller Additions
+  { 304               , XBMCK_RETURN      }, // A
+  { 305               , XBMCK_ESCAPE      }, // B
```
Comment (FernetMenta) [inline xbmc/input/linux/LinuxInputDevices.cpp]: we have to stop those hacks. symbols are defined in input-event-codes. 305 is already defined: https://github.com/torvalds/linux/blob/master/include/uapi/linux/input-event-codes.h#L382

**Comment by MartijnKaijser** [comment]
@garbear ping

**Comment by garbear** [comment]
Looks OK to me. Eventually this code will probably be replaced by libinput.

**Comment by FernetMenta** [comment]
It does not look ok! See my comment

**Comment by garbear** [comment]
Oh I missed fernet's comment because it was collapsed, yes these magic numbers should be replaced.

**Comment by FernetMenta** [comment]
And we must not redefine codes like 305.

**Comment by Rechi** [comment]
Is this obsolete with #13176?

**Comment by MartijnKaijser** [comment]
@garbear @lrusak ping

**Comment by Rechi** [comment]
I will close this PR. If there's interest please open a new one (with required changes if needed).

**Pull request closed without merging**